runtime/jvmti/jvmtiThread.cpp

```diff
@@ -1179,11 +1183,27 @@ resumeThread(J9VMThread *currentThread, jthread thread)
 #if JAVA_SPEC_VERSION >= 19
 		J9JavaVM *vm = currentThread->javaVM;
 		j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
+		U_64 internalSuspendState = J9OBJECT_U64_LOAD(currentThread, threadObject, vm->internalSuspendStateOffset);
+		U_64 carrierVMThread = (internalSuspendState & J9_VIRTUALTHREAD_INTERNAL_STATE_CARRIERID_MASK);
+
+		/* Update internal flag before clearing flag to avoid sync issue. */
+		if (J9_ARE_NO_BITS_SET(internalSuspendState, J9_VIRTUALTHREAD_INTERNAL_STATE_SUSPENDED)) {
+			rc = JVMTI_ERROR_THREAD_NOT_SUSPENDED;
```

Comment:
the current approach only loads the field once. but, it adds another location to maintain if the code related to the bit-mask changes. if we use the helpers here, there will be an extra load, but the code will be easy to maintain since the bit-mask operations will only be done from the helpers.

```suggestion
		J9VMThread *carrierVMThread = VM_VMHelpers::getCarrierVMThread(currentThread, threadObject);

		/* Update internal flag before clearing flag to avoid sync issue. */
		if (!VM_VMHelpers::isThreadSuspended(currentThread, threadObject)) {
			rc = JVMTI_ERROR_THREAD_NOT_SUSPENDED;
```